pettingzoo/classic/checkers/checkers.py

```diff
@@ -0,0 +1,608 @@
+        moves = self.ch.legal_moves()
+        legal_moves = []
+        for move in moves:
+            srcpos = self._act_rel_to_abs(move[0])
+            destpos = self._act_rel_to_abs(move[1])
+
+            direction = -1
+            if destpos == srcpos - 9 or destpos == srcpos - 18:
+                direction = 0
+            elif destpos == srcpos - 7 or destpos == srcpos - 14:
+                direction = 1
+            elif destpos == srcpos + 7 or destpos == srcpos + 14:
+                direction = 2
+            elif destpos == srcpos + 9 or destpos == srcpos + 18:
+                direction = 3
+
+            legal_moves.append(self._act_abs_to_rel(srcpos) + (64 * direction))
+
+        return legal_moves
+
+    def step(self, action, observe=True):
+
+        if action not in self.legal_moves():
+            warnings.warn(
+                "Bad checkers move made, game terminating with current player losing. \n env.infos[player]['legal_moves'] contains a list of all legal moves that can be chosen."
+            )
+            winner = "white" if self.last_turn == "black" else "black"
+        else:
+            self.num_moves += 1
+            action = self._parse_action(action)
+            self.board, turn, last_moved_piece, moves, winner = self.ch.move(
+                action[0], action[1]
+            )
+
+            self.agent_selection = self._agent_selector.next()
+
+            if turn == self.last_turn:
+                self.agent_selection = self._agent_selector.next()
+                self.agent_order.reverse()
+            self.last_turn = turn
+
+            self.observation = self._read_observation()
+
+            # self.ch.print_board()
+            # print(self.agent_selection)
+            # print(self.observe(self.agent_selection))
+            # print(self.observation)
+            # self.infos[self.agent_selection]['legal_moves']
+
+        self.infos[self.agent_selection]["legal_moves"] = self.legal_moves()
+
+        if winner is None and self.num_moves > self.num_moves_max:
+            print("Draw")
+            self.winner = -1
+            self.rewards[self.agents[0]] = 0
+            self.rewards[self.agents[1]] = 0
+        else:
+            if winner == "black":
+                self.winner = 0
+                self.rewards[self.agents[0]] = 1
+                self.rewards[self.agents[1]] = -1
+            elif winner == "white":
+                self.winner = 1
+                self.rewards[self.agents[0]] = -1
+                self.rewards[self.agents[1]] = 1
+            else:
+                pass
+
+        self.dones[self.agent_order[0]] = winner is not None
+        self.dones[self.agent_order[1]] = winner is not None
+
+        if observe:
+            next_observation = self.observe(self.agent_selection)
+        else:
+            next_observation = None
+        return next_observation
+
+    def render(self, mode="human"):
+        print(self.ch.flat_board())

```

Comment:
Perhaps a clearer render would just to print it like this?

```
  C  _  _  _
_  K  _  _
  _  K  C  _
_  _  _  _
  _  _  _  _
_  _  c  _
  k  _  _  _
c  _  _  _
```
where c is a checker and k is a king, and capital letters represent white? This is modeled after how python-chess does it, but there may be some other reasonable way.

Reply:
I can change it to something like this

Reply:
![Screenshot from 2020-10-07 02-22-28](https://user-images.githubusercontent.com/7774618/95294880-0c2ce800-0844-11eb-8014-6e8442f7f44c.png)

Reply:
So you don't have to print out the reward with the observation, as the user can easily print the reward themselves. 

Otherwise looks good.

Reply:
That was just my testing script, it's not part of the render function